lib/ansible/executor/task_result.py

```diff
@@ -69,3 +72,33 @@ def _check_key(self, key):
                 if isinstance(res, dict):
                     flag |= res.get(key, False)
             return flag
+
+    def clean_copy(self):
+
+        ''' returns 'clean' taskresult object '''
+
+        # FIXME: clean task_fields, _task and _host copies
+        result = TaskResult(self._host, self._task, {}, self._task_fields)
+
+        # statuses are already reflected on the event type
+        if result._task and result._task.action in ['debug']:
+            # debug is verbose by default to display vars, no need to add invocation
+            ignore = _IGNORE + ('invocation',)
+        else:
+            ignore = _IGNORE
+
+        if self._result.get('_ansible_no_log', False):
```

Comment:
since this is all for dealing with a TaskResult._result, move the _result processing code to it's own method and call from here?
``` python
def _clean_result(self, result, ignore):
<...>
```

It could even be staticmethod or module scope method
``` python
def clean_result_dict(result_dict, ignore):
   <..>
   return result_dict
```

so, clean_result() could:

``` python

    if self._result:
       result._result = clean_result(result._result, ignore)
    return result
```

Reply:
self._result is the start, planning on also cleaning up task and play